### PR TITLE
Basic report framework

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

File: daq/runner.py [hunk 1]
```diff
@@ -1075,6 +1078,13 @@ def _send_device_result(self, device, test_results):
         except Exception as e:
             LOGGER.error("Failed to send device results for device %s: %s ", device, e)
 
+    def report_sink(self, report_dict):
```
Comment (henry54809) [inline daq/runner.py:1081]: If it's not consumed by some grpc client, does it really have to be in the protobuf? The existing dict / a dataclass to replace https://github.com/faucetsdn/daq/blob/master/daq/report.py#L240 should suffice.

**Comment by grafnu** [comment]
It will be consumed by some client, even if it's not gRPC (protobuf isn't
only useful for grpc). I want something that is well-structured and
cross-language, so something more than an arbitrary dict (unstructured),
and not just a dataclass (python-only).

**Comment by grafnu** [comment]
Any additional comments?

File: proto/report.proto [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+message TestModule {
+  // Line items test results from this module.
+  map<string, TestItem> tests = 1;
+}
+
+message TestItem {
+  // Test result (pass/fail)
+  string result = 1;
```
Comment (henry54809) [inline proto/report.proto:21]: Should that be in enum? But we can revisit this later.

**Review by henry54809 — APPROVED** [review]

**Pull request merged**